Lib/pydoc.py

```diff
@@ -2614,18 +2614,50 @@ def browse(port=0, *, open_browser=True, hostname='localhost'):
 def ispath(x):
     return isinstance(x, str) and x.find(os.sep) >= 0
 
+def _get_revised_path(given_path, argv0):
+    """Ensures current directory is on returned path, and argv0 directory is not
+
+    Exception: argv0 dir is left alone if it's also pydoc's directory.
+
+    Returns a new path entry list, or None if no adjustment is needed.
+    """
+    # Scripts may get the current directory in their path by default if they're
+    # run with the -m switch, or directly from the current directory.
+    # The interactive prompt also allows imports from the current directory.
+
+    # Accordingly, if the current directory is already present, don't make
+    # any changes to the given_path
+    if '' in given_path or os.curdir in given_path or os.getcwd() in given_path:
```

Comment:
In what circumstances `sys.path` contains `''` or `'.'`?

Reply:
`''` will be present for `-c` and the interactive prompt, `'.'` will only be present if you added it explicitly (originally it was added by `pydoc.cli()` itself, but I changed that as part of this PR)

Reply:
What if just remove the whole `sys.path` patching code? Both `./python -m pydoc` and `./python Lib/pydoc.py` looks working after this.

Reply:
The problem the path manipulation solves is that `pydoc` itself needs to be able to find modules in the current directory (where the user's own code is likely to be), even when run via a wrapper script installed as `/usr/bin/pydoc` (or the Windows equivalent).

I'm not sure there's a regression test for that, though.